src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -194,30 +194,23 @@ class BookingExtractionService(
       dates[PRRD] = latestReleaseDate
     }
     if (booking.sentences.any { it is DetentionAndTrainingOrderSentence }) {
-      dates[MTD] = latestReleaseDate
-      calculateWhenAllDtos(mostRecentSentenceByExpiryDate, dates)
-      if (sentences.size > 1 && !sentences.any { it is ConsecutiveSentence } && !sentences.all { it is DetentionAndTrainingOrderSentence }) {
-        val firstNonDtoSentence = sentences.first { it !is DetentionAndTrainingOrderSentence }
-        val firstDtoSentence = sentences.first { it is DetentionAndTrainingOrderSentence }
-        val type = if (firstNonDtoSentence.releaseDateTypes.contains(CRD)) CRD else ARD
-        dates[type] = firstNonDtoSentence.sentenceCalculation.releaseDate
-        val midTermDate = if (firstDtoSentence.sentenceCalculation.isImmediateRelease() && firstDtoSentence.identificationTrack == SentenceIdentificationTrack.DTO_AFTER_PCSC) {
-          firstDtoSentence.sentencedAt
-        } else if (type == CRD) {
-          if (firstDtoSentence.identificationTrack == SentenceIdentificationTrack.DTO_BEFORE_PCSC) {
-            firstDtoSentence.sentenceCalculation.unadjustedDeterminateReleaseDate
-          } else {
-            firstDtoSentence.sentenceCalculation.adjustedDeterminateReleaseDate
-          }
-        } else {
-          latestReleaseDate
-        }
+      if (booking.sentences.all { it.isDto() }) {
+        dates[MTD] = latestReleaseDate
+        calculateWhenAllDtos(mostRecentSentenceByExpiryDate, dates)
+
+      } else {
+        val earliestNonDtoSentence = sentences.sortedBy { it.sentencedAt }.first { !it.isDto() }
```

Comment:
I can't see in the ticket what the logic should be. Should it be earliest by sentence date? Normally its latest by release date.

Reply:
Yes that's right tbf. It only worked because there are only two sentences.